Compilation warning/error with clang

**Describe the bug**

1. : 

```
In file included from /GEOSX/src/coreComponents/constitutive/fluid/BlackOilFluid.cpp:15:
/GEOSX/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp:1071:40: error: argument 'dRs_dComp' of type 'const geosx::real64[2]' (aka 'const double[2]') with mismatched bound [-Werror,-Warray-parameter]
                          real64 const dRs_dComp[HNC_BO],
                                       ^
/GEOSX/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp:293:47: note: previously declared as 'const geosx::real64[]' (aka 'const double[]') here
                                 real64 const dRs_dComp[],
                                              ^
/GEOSX/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp:1074:40: error: argument 'dBo_dComp' of type 'const geosx::real64[2]' (aka 'const double[2]') with mismatched bound [-Werror,-Warray-parameter]
                          real64 const dBo_dComp[HNC_BO],
                                       ^
/GEOSX/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp:296:47: note: previously declared as 'const geosx::real64[]' (aka 'const double[]') here
                                 real64 const dBo_dComp[],

```

2. I didn't copy the error msg but clang complains also with this unused counter  

**To Reproduce**
Compile using:
```
➜  ~ clang --version
clang version 15.0.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

```

**Expected behavior**
Compilation should proceed...

**Screenshots**
n.a.

**Platform (please complete the following information):**
- Machine [personal] 
- Compiler: [clang 15.0.7]
- GEOSX Version [develop branch]

**Additional context**
n.a.

--- a/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp
+++ b/src/coreComponents/constitutive/fluid/BlackOilFluid.hpp
@@ -290,10 +290,10 @@ private:
                                  bool const useMass,
                                  real64 const Rs,
                                  real64 const dRs_dPres,
-                                 real64 const dRs_dComp[],
+                                 real64 const dRs_dComp[HNC_BO],
                                  real64 const Bo,
                                  real64 const dBo_dPres,
-                                 real64 const dBo_dComp[],
+                                 real64 const dBo_dComp[HNC_BO],
                                  real64 & dens,
                                  arraySlice1d< real64, multifluid::USD_PHASE_DC - 3 > const & dDens ) const;
 

--- a/src/coreComponents/unitTests/constitutiveTests/testCO2BrinePVTModels.cpp
+++ b/src/coreComponents/unitTests/constitutiveTests/testCO2BrinePVTModels.cpp
@@ -505,7 +505,7 @@ TEST_F( EzrokhiBrineViscosityTest, brineViscosityValuesAndDeriv )
 
   EzrokhiBrineViscosity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  integer counter = 0;
+  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -513,7 +513,7 @@ TEST_F( EzrokhiBrineViscosityTest, brineViscosityValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, true, eps, relTol );
-        counter++;
+        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -653,7 +653,7 @@ TEST_F( PhillipsBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
 
   PhillipsBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  integer counter = 0;
+  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -661,7 +661,7 @@ TEST_F( PhillipsBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, false, eps, relTol );
-        counter++;
+        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -704,7 +704,7 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMassValuesAndDeriv )
 
   EzrokhiBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  integer counter = 0;
+  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -712,7 +712,7 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMassValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, true, eps, relTol );
-        counter++;
+        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -735,7 +735,7 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
 
   EzrokhiBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  integer counter = 0;
+  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -743,7 +743,7 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, false, eps, relTol );
-        counter++;
+        // counter++;
       }
     }
     comp[0] += deltaComp;

--- a/src/coreComponents/unitTests/constitutiveTests/testCO2BrinePVTModels.cpp
+++ b/src/coreComponents/unitTests/constitutiveTests/testCO2BrinePVTModels.cpp
@@ -505,7 +505,6 @@ TEST_F( EzrokhiBrineViscosityTest, brineViscosityValuesAndDeriv )
 
   EzrokhiBrineViscosity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -513,7 +512,6 @@ TEST_F( EzrokhiBrineViscosityTest, brineViscosityValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, true, eps, relTol );
-        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -653,7 +651,6 @@ TEST_F( PhillipsBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
 
   PhillipsBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -661,7 +658,6 @@ TEST_F( PhillipsBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, false, eps, relTol );
-        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -704,7 +700,6 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMassValuesAndDeriv )
 
   EzrokhiBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -712,7 +707,6 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMassValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, true, eps, relTol );
-        // counter++;
       }
     }
     comp[0] += deltaComp;
@@ -735,7 +729,6 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
 
   EzrokhiBrineDensity::KernelWrapper pvtFunctionWrapper = pvtFunction->createKernelWrapper();
 
-  // integer counter = 0;
   for( integer iComp = 0; iComp < 3; ++iComp )
   {
     for( integer iPres = 0; iPres < 3; ++iPres )
@@ -743,7 +736,6 @@ TEST_F( EzrokhiBrineDensityTest, brineCO2DensityMolarValuesAndDeriv )
       for( integer iTemp = 0; iTemp < 3; ++iTemp )
       {
         testNumericalDerivatives( pvtFunctionWrapper, P[iPres], TC[iTemp], comp, false, eps, relTol );
-        // counter++;
       }
     }
     comp[0] += deltaComp;

